### PR TITLE
Send labels and annotations

### PR DESCRIPTION
**Pull request by matas-cast**
Collect selected labels and annotations from CRI API and send them with runtime events

### CODE REVIEW

File: cmd/agent/daemon/daemon.go [hunk 1]
```diff
@@ -105,6 +105,10 @@ func NewRunCommand(version string) *cobra.Command {
 		automountCgroupv2 = command.Flags().Bool("automount-cgroupv2", true, "Automount cgroupv2 if not mounted")
 
 		redactSensitiveValuesRegexStr = command.Flags().String("redact-sensitive-values-regex", "", "Regex which will be used to detect sensitive values in process exec args")
+		
+		criEndpoint          = command.Flags().String("cri-endpoint", "unix:///run/containerd/containerd.sock", "Cri endpoint")
```
Comment (patrickpichler) [inline cmd/agent/daemon/daemon.go:109]: Very important nitpick: `CRI endpoint`

File: pkg/containers/client.go [hunk 2]
```diff
@@ -171,6 +193,35 @@ func (c *Client) addContainerWithCgroup(container containerdContainers.Container
...
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sandbox.Items) > 0 {
+		for k, v := range sandbox.Items[0].Labels {
+			if _, ok := c.forwardedLabels[k]; ok {
```
Comment (patrickpichler) [inline pkg/containers/client.go:208]: I think it would be beneficial to also support wildcards.

File: pkg/containers/client.go [hunk 3]
```diff
@@ -171,6 +193,35 @@ func (c *Client) addContainerWithCgroup(container containerdContainers.Container
...
+		Filter: &criapi.PodSandboxFilter{
+			Id: container.SandboxID,
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sandbox.Items) > 0 {
```
Comment (patrickpichler) [inline pkg/containers/client.go:206]: I think in the `else` we can return an error, as we only support kubernetes pods anyway.

Comment (patrickpichler) [reply pkg/containers/client.go:206]: Hm could this create a race condition though, so maybe not do this. I wonder if we can get an event stream from the CRI.

File: pkg/containers/client.go (hunk 2 above)
Comment (anjmao) [reply pkg/containers/client.go:208]: As minimum it should contain prefix

File: cmd/agent/daemon/daemon.go [hunk 4]
```diff
@@ -105,6 +105,10 @@ func NewRunCommand(version string) *cobra.Command {
...
+		ebpfEventLabels      = command.Flags().StringSlice("ebpf-events-include-pod-labels", []string{}, "List of label keys to be sent with eBPF events, separated by comma")
```
Comment (anjmao) [inline cmd/agent/daemon/daemon.go:110]: Extend e2e test and assert that event contain labels and annotations

File: pkg/containers/client.go (hunk 2 above)
Comment (matas-cast) [reply pkg/containers/client.go:208]: With prefix - simple map lookup won't work, will need to match with all labels slice

File: pkg/containers/client.go (hunk 3 above)
Comment (matas-cast) [reply pkg/containers/client.go:206]: Don't see anything similar to event stream. 

But what race condition you're thinking of, and how an event stream would help?

**Review by anjmao — APPROVED** [review]

**Pull request merged**